components/admin/AddCaseAddress.tsx

```diff
@@ -141,7 +143,7 @@ export default function AddCaseAddress({
           <SummaryListActions wideActions={true}>
             {addresses
               ? addresses.map((addressItem, index) => (
-                  <FormGroup key={index}>
+                  <FormGroup key={uniqueID()}>
```

Comment:
Introducing an extra NPM library for this is overkill, you could just do:
```ts
addresses.map((addressItem, index) => (
                  <FormGroup key={`address-item-${index}`}>
                     ...
```
and that should be unique enough. :thinking:

Reply:
I've gone unique (without a huge amount of consideration to each case) is it's the recommended approach in react rather than using array index that might change/reorder be filtered etc.

Take your point on avoiding adding another package - I might take a look and see if any of the packages we currently use have a generator. But uuid pretty solid package.

Reply:
Unless your elements change order between re-renders, you probably don't need a random key,
but if you do want a random one anyway, then you don't need any package in general - all you need should be in the base node.js already as part of Math library:

```js
const randKey = () => Math.floor(Math.random() * 10e6)
```

Reply:
Yeah, agree it's not strictly necessary in many cases, but it's probably good practice and a bit safer. 

Re using built in math - this generally seems pretty discouraged / seen as an unstable approach.

Reply:
Seems like the react doesn't recommend any randomness in the keys.
They only say you should use `uuid` if your items or data you're referring to is persisted .. to say local storage.

| ![image](https://github.com/user-attachments/assets/c4453066-3da5-44d0-85b1-1c2ac6039d0a) |
| --- |
| Snippet from react docs |

Reply:
good shout - I needed to read the documentation properly. To be fair, a tonne of conflicting stuff around the approach. But basically the ID should stable, irrelevant of re-renders, which makes sense. 

Will go back through this for a tidy, but it seems like in most cases the approach will have to be the index plus a relevant value. Thanks :)